partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -884,8 +884,12 @@ internal class PlanTyper(private val env: Env, config: Context) {
         private fun replaceCaseBranch(originalRex: Rex, outputType: CompilerType, function: DynamicTyper.Mapping): Rex {
             return when (function) {
                 is DynamicTyper.Mapping.Coercion -> {
-                    val cast = env.resolveCast(originalRex, function.target)!!
-                    Rex(outputType, cast)
+                    if (originalRex.type == function.target) {
```

Comment:
self-review: we were adding extra `CAST`s which weren't necessary (e.g. when the `CASE WHEN` branches had the same type). e.g. 

```
CASE WHEN ... THEN 1 ELSE 2 END
```

would get typed as

```
CASE WHEN ... THEN CAST(1 AS INT) ELSE CAST(1 AS INT) END
```